From cddff0b24e294496853d465d0ab3af853ed952a2 Mon Sep 17 00:00:00 2001
From: Adam Stolcenburg <adam_stolcenburg@comcast.com>
Date: Tue, 14 Jan 2025 12:23:18 +0100
Subject: [PATCH 1/5] PREMIUMAPP-3142 Change keymap to better reflect RDK RCUs

---
 src/device/rdk/interface.rs | 14 +++-----------
 1 file changed, 3 insertions(+), 11 deletions(-)

diff --git a/src/device/rdk/interface.rs b/src/device/rdk/interface.rs
index e242cd1..f02c070 100644
--- a/src/device/rdk/interface.rs
+++ b/src/device/rdk/interface.rs
@@ -355,23 +355,15 @@ lazy_static! {
         keycode_map.insert(String::from("KEY_VOLUME_UP"),175);
         keycode_map.insert(String::from("KEY_VOLUME_DOWN"),174);
         keycode_map.insert(String::from("KEY_MUTE"),173);
-        keycode_map.insert(String::from("KEY_EXIT"),27);
         keycode_map.insert(String::from("KEY_UP"),38);
-        keycode_map.insert(String::from("KEY_PAGE_UP"),33);
-        keycode_map.insert(String::from("KEY_PAGE_DOWN"),34);
         keycode_map.insert(String::from("KEY_RIGHT"),39);
         keycode_map.insert(String::from("KEY_DOWN"),40);
         keycode_map.insert(String::from("KEY_LEFT"),37);
         keycode_map.insert(String::from("KEY_ENTER"),13);
         keycode_map.insert(String::from("KEY_BACK"),8);
-        keycode_map.insert(String::from("KEY_PLAY"),179);
-        keycode_map.insert(String::from("KEY_PLAY_PAUSE"),179);
-        keycode_map.insert(String::from("KEY_PAUSE"),179);
-        keycode_map.insert(String::from("KEY_STOP"),178);
-        keycode_map.insert(String::from("KEY_REWIND"),227);
-        keycode_map.insert(String::from("KEY_FAST_FORWARD"),228);
-        keycode_map.insert(String::from("KEY_SKIP_REWIND"),177);
-        keycode_map.insert(String::from("KEY_SKIP_FAST_FORWARD"),176);
+        keycode_map.insert(String::from("KEY_PLAY_PAUSE"),227);
+        keycode_map.insert(String::from("KEY_REWIND"),224);
+        keycode_map.insert(String::from("KEY_FAST_FORWARD"),223);
         keycode_map.insert(String::from("KEY_0"),48);
         keycode_map.insert(String::from("KEY_1"),49);
         keycode_map.insert(String::from("KEY_2"),50);
-- 
2.43.0

